runtime/src/chplsys.c

```diff
@@ -85,12 +85,12 @@ size_t chpl_getHeapPageSize(void) {
     else {
 
       size_t tmpPageSize;
-      int  num_scanned;
-      char units;
+      int scanCnt;
+      char units[4]; // leave room for terminating null byte
 
-      if ((num_scanned = sscanf(ev, "%zi%c", &tmpPageSize, &units)) != 1) {
-        if (num_scanned == 2 && strchr("kKmMgG", units) != NULL) {
-          switch (units) {
+      if ((scanCnt = sscanf(ev, "%zd%1[kKmMgG]", &tmpPageSize, units)) > 0) {
```

Comment:
I should have commented on this before.  I can't find any authoritative documentation about the format of the HUGETLB_DEFAULT_PAGE_SIZE environment variable.  The original code allowed for the possibility that it might be just a number with no suffix, but if there was a suffix it had to directly follow the number with no intervening white space.  The new code requires a suffix but allows white space between the number and suffix.  If there is definitive documentation about the format let's follow it; otherwise let's allow both the intervening white space and a number without a suffix.